torch/utils/data/dataloader.py

```diff
@@ -94,7 +94,21 @@ def is_alive(self):
             return not self.manager_dead
 
 
-def _worker_loop(dataset, index_queue, data_queue, done_event, collate_fn, seed, init_fn, worker_id):
+def _fix_dataset(dataset, base_seed):
+    prev_getitem = dataset.__getitem__
+
+    def getitem(self, idx):
+        random.seed(base_seed + idx)
```

Comment:
Here you are assuming that index is an integer.

Reply:
Actually, idx doesn't have to be an integer. See the result of the code below

```python
>> random.seed(1.1);print(random.random())
0.22415427849366876
>> random.seed(1.1);print(random.random())
0.22415427849366876
>>random.seed(1.2);print(random.random())
0.24751702373640583
```

According to the official python docs

> If a(seed) is not None or an int or a long, then hash(a) is used instead